Wikipedia/Code/UIViewController+WMFOpenExternalUrl.m

```diff
@@ -19,10 +19,10 @@ - (void)wmf_openExternalUrl:(NSURL *)url useSafari:(BOOL)useSafari {
     NSParameterAssert(url);
 
     //If zero rated, don't open any external (non-zero rated!) links until user consents!
-    if ([SessionSingleton sharedInstance].zeroConfigState.disposition && [[NSUserDefaults wmf_userDefaults] boolForKey:@"ZeroWarnWhenLeaving"]) {
-        WMFZeroMessage *zeroMessage = [SessionSingleton sharedInstance].zeroConfigState.zeroMessage;
-        NSString *exitDialogTitle = zeroMessage.exitTitle ?: MWLocalizedString(@"zero-interstitial-title", nil);
-        NSString *messageWithHost = [NSString stringWithFormat:@"%@\n\n%@", zeroMessage.exitWarning ?: MWLocalizedString(@"zero-interstitial-leave-app", nil), url.host];
+    if ([SessionSingleton sharedInstance].zeroConfigurationManager.isZeroRated && [[NSUserDefaults wmf_userDefaults] boolForKey:WMFZeroWarnWhenLeaving]) {
```

Comment:
2 comments that both point to similar conclusions:

1.  "Tell don't ask" - object oriented design usually takes the form of this statement.  For example, If you are talking to a car object:
"Car are you unlocked? if so, then open the door, if not put the key in"  or, 
"Unlock, open door"

In the first example you are examining the state of an object and making decisions for it. In the second example you are letting the object that owns the state make decisions it knows about because it should have all the info it needs to know to perform the action.

2. This category is called `WMFOpenExternalURL` but there is a lot of Zero logic in here. A developer would probably not think to look in this file for such logic. 


Both of these point to putting this logic in the configuration manager - which is the object responsible for managing the zero state.